pkg/analyzer/analyzers/slack/slack.go

```diff
@@ -12,8 +15,86 @@ import (
 	"github.com/jedib0t/go-pretty/table"
 	"github.com/trufflesecurity/trufflehog/v3/pkg/analyzer/analyzers"
 	"github.com/trufflesecurity/trufflehog/v3/pkg/analyzer/config"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/analyzer/pb/analyzerpb"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/context"
 )
 
+var _ analyzers.Analyzer = (*Analyzer)(nil)
+
+type Analyzer struct {
+	Cfg *config.Config
+}
+
+func (Analyzer) Type() analyzerpb.AnalyzerType { return analyzerpb.AnalyzerType_Slack }
+
+func (a Analyzer) Analyze(_ context.Context, credInfo map[string]string) (*analyzers.AnalyzerResult, error) {
+	key, ok := credInfo["key"]
+	if !ok {
+		return nil, errors.New("key not found in credentialInfo")
+	}
+
+	info, err := AnalyzePermissions(a.Cfg, key)
+	if err != nil {
+		return nil, err
+	}
+	return secretInfoToAnalyzerResult(info), nil
+}
+
+func secretInfoToAnalyzerResult(info *SecretInfo) *analyzers.AnalyzerResult {
+	if info == nil {
+		return nil
+	}
+	result := analyzers.AnalyzerResult{
+		AnalyzerType: analyzerpb.AnalyzerType_Slack,
+		Metadata:     nil,
+	}
+
+	resourceType := "user"
+	fullyQualifiedName := info.User.UserId
+	if info.User.BotId != "" {
+		resourceType = "bot"
+		fullyQualifiedName = info.User.BotId
+	}
+	resource := analyzers.Resource{
+		Name:               info.User.User,
+		FullyQualifiedName: fullyQualifiedName,
+		Type:               resourceType,
+		Metadata: map[string]any{
+			"url":     info.User.Url,
+			"team":    info.User.Team,
+			"team_id": info.User.TeamId,
+			"scopes":  strings.Split(info.Scopes, ","),
+		},
+	}
+
+	// extract all permissions
+	permissions := extractPermissions(info)
+
+	result.Bindings = analyzers.BindAllPermissions(resource, permissions...)
+
+	return &result
+}
+
+func extractPermissions(info *SecretInfo) []analyzers.Permission {
+	var permissions []analyzers.Permission
+
+	for _, scope := range strings.Split(info.Scopes, ",") {
+		perms, ok := scope_mapping[scope]
```

Comment:
Does `scope_mapping` use the generated permission type? I would feel better about this if it did, to make sure we're assigning permissions from the enumerated set.